src/fair/constraint.py

```diff
@@ -221,9 +221,13 @@ def from_item_lists(
         b = dok_array((rows, 1), dtype=np.int_)
 
         for i in range(rows):
-            for value in preferred_values[i]:
+            for values in preferred_values[i]:
                 for item in schedule:
-                    if item.value(preferred_feature) == value:
+                    match = True
+                    for j, feature in enumerate(preferred_features):
```

Comment:
What is j iterating over exactly? how do you know `values[j]` won't go out of range? 
I'm having issues with this line while running the course allocation data repo scripts because of this. 
I found that j usually takes values 0 or 1, so whenever the number of preferred_values[i] is 0 or 1, the code crashes. 

I tried doing this:
`                        if len(values)==0:
                            match = False
                        elif item.value(feature) != values[j]:
                            match = False`
But did not fix it, since it only recovers the case in which the preferred_values=[] (empty preferences), but it could still happen that j is out of range for values (values has 1 element and j is 1). Is it enough to just do something like this?
`                        if len(values)<=j:
                            match = False
                        elif item.value(feature) != values[j]:
                            match = False`